lib/ansible/modules/cloud/amazon/ec2_vpc_vgw.py

```diff
@@ -376,40 +369,33 @@ def ensure_vgw_present(client, module):
     params['Tags'] = module.params.get('tags')
     params['VpnGatewayIds'] = module.params.get('vpn_gateway_id')
 
-    # Check that a name argument has been supplied.
-    if not module.params.get('name'):
-        module.fail_json(msg='A name is required when a status of \'present\' is suppled')
+    # check that the vpc_id exists. If not, an exception is thrown
+    if params['VpcId']:
+        vpc = find_vpc(client, module)
 
     # check if a gateway matching our module args already exists
     existing_vgw = find_vgw(client, module)
 
-    if existing_vgw != [] and existing_vgw[0]['State'] != 'deleted':
+    if existing_vgw != []:
         vpn_gateway_id = existing_vgw[0]['VpnGatewayId']
         vgw, changed = check_tags(client, module, existing_vgw, vpn_gateway_id)
 
         # if a vpc_id was provided, check if it exists and if it's attached
         if params['VpcId']:
 
-            # check that the vpc_id exists. If not, an exception is thrown
-            vpc = find_vpc(client, module)
             current_vpc_attachments = existing_vgw[0]['VpcAttachments']
 
             if current_vpc_attachments != [] and current_vpc_attachments[0]['State'] == 'attached':
-                if current_vpc_attachments[0]['VpcId'] == params['VpcId'] and current_vpc_attachments[0]['State'] == 'attached':
-                    changed = False
-                else:
-
+                if not (current_vpc_attachments[0]['VpcId'] == params['VpcId'] and current_vpc_attachments[0]['State'] == 'attached'):
```

Comment:
This might be nicer as `if current_vpc_attachments[0]['VpcId'] != params['VpcId'] or current_vpc_attachments[0]['State'] != 'attached':`